### PR TITLE
Complete doxygen documentation

### PR DESCRIPTION
**Pull request by morxa**
@snoato has recently pointed out that the doxygen docs are incomplete, even though we configured doxygen to error out if a warning occurs. This check no longer worked due to a syntax change in a some doxygen update.

Re-enabled the error and also complete the documentation. While at it, also add documentation for namespaces that have not been completely documented before.

### CODE REVIEW

**Comment by morxa** [comment]
Please ignore the codecov error, there was (almost) no actual code change.

File: src/visualization/include/visualization/interactive_tree_to_graphviz.h [hunk 1]
```diff
@@ -6,33 +6,35 @@
...
 void print_interactive_help();
 
+/// Possible modes of the interactive debugger.
 enum class Mode {
-	NAVIGATE,
-	INSERT,
-	INSERT_AND_FOLLOW,
+	NAVIGATE,          /// Move around in the search tree.
+	INSERT,            /// Insert a child node but do not change the selected node.
+	INSERT_AND_FOLLOW, /// Insert a childe and select it.
```
Comment (snoato) [inline src/visualization/include/visualization/interactive_tree_to_graphviz.h:26]: remove the e

File: src/search/include/search/adapter.h [hunk 2]
```diff
@@ -26,7 +27,12 @@ template <typename Plant,
 class get_next_canonical_words
 {
 public:
-	get_next_canonical_words(const std::set<ActionType> & = {}, const std::set<ActionType> & = {})
+	/** Construct the comparator with the given action partitioning.
+	 * @param controller_actions The actions that the controller can select
+	 * @param environment_actions The actions that the environment can select
+	 */
+	get_next_canonical_words([[maybe_unused]] const std::set<ActionType> &controller_actions  = {},
+	                         [[maybe_unused]] const std::set<ActionType> &environment_actions = {})
```
Comment (snoato) [inline src/search/include/search/adapter.h:35]: Should this warning be addressed in this PR or in a separate one?

Comment (morxa) [reply src/search/include/search/adapter.h:35]: This warning can be ignored, the function body is empty anyway, as this is just the base class.

**Comment by morxa** [comment]
Thanks for the thorough checks! The issues you have pointed out should be fixed now.

**Review by snoato — APPROVED** [review]
Thank you!

**Pull request merged**